bin/pyenv-win-venv.ps1

```diff
@@ -44,53 +48,68 @@ function  main {
             if (test-path $python_version_file) {
                 $env_name = (Get-Content $python_version_file)
                 if ($env_name -And (test-path -PathType container "$app_env_dir\$env_name")) {
-                    &"$app_env_dir\$env_name\Scripts\Activate.ps1" 
+                    if ($subcommand1 -eq "ps1") {
+                        &"$app_env_dir\$env_name\Scripts\Activate.ps1" 
+                    }
+                    else {
+                        cmd /k "$app_env_dir\$env_name\Scripts\activate.bat"
+                    } 
                 }
             }
         }
     }
-    elseif ($subcommand1 -eq "activate") {
-        if (!$subcommand2) {
+    elseif ($subcommand2 -eq "activate") {
+        if (!$subcommand3) {
             HelpActivate
         }
-        elseif (test-path -PathType container "$app_env_dir\$subcommand2") {
-            &"$app_env_dir\$subcommand2\Scripts\Activate.ps1"
+        elseif (test-path -PathType container "$app_env_dir\$subcommand3") {
+            if ($subcommand1 -eq "ps1") {
+                &"$app_env_dir\$subcommand3\Scripts\Activate.ps1" 
+            }
+            else {
+                cmd /k "$app_env_dir\$subcommand3\Scripts\activate.bat"
+            }
             
         }
         else {
-            Write-Host "Env: $subcommand2 is not installed. Install using `"pyenv-win-venv install <python_version> $subcommand2"`"
+            Write-Host "Env: $subcommand3 is not installed. Install using `"pyenv-win-venv install <python_version> $subcommand3"`"
         }
     }
-    elseif ($subcommand1 -eq "deactivate") {
-        deactivate
+    elseif ($subcommand2 -eq "deactivate") {
+        if ($subcommand1 -eq "ps1") {
```

Comment:
Might want to do a check if `$env:VIRTUAL_ENV` is available (standard way to check for active venv) so that you can print your own error.

If no env is active you get a bunch of red text on powershell and a different error on cmd